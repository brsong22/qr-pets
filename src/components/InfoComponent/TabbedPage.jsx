import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';

class TabbedPage extends Component {
  constructor(props) {
    super(props);

    this.state = {
      onTab: 0,
    };
    this.handleTabClick = this.handleTabClick.bind(this);
  }

  handleTabClick(_, tabIndex) {
    this.setState({ onTab: tabIndex });
  }

  renderInfoText() {
    const { info } = this.props;
    const { onTab } = this.state;

    return (<p>{info[onTab]}</p>);
  }

  render() {
    const { tabs } = this.props;
    const { onTab } = this.state;
    return (
      <div className="tabbedContainer">
        <Tabs value={onTab} onChange={this.handleTabClick}>
          {tabs.map(tab => <Tab key={tab} label={tab} />)}
        </Tabs>
        {this.renderInfoText()}
      </div>
    );
  }
}

TabbedPage.propTypes = {
  tabs: PropTypes.arrayOf(PropTypes.string),
  info: PropTypes.arrayOf(PropTypes.string),
};

TabbedPage.defaultProps = {
  tabs: [],
  info: [],
};

export default TabbedPage;
