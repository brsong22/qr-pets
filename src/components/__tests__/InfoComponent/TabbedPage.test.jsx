
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TabbedPage from '../../InfoComponent/TabbedPage';

describe('TabbedPage', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<TabbedPage />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
