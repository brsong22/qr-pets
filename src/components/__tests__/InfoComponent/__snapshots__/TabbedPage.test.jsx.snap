// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabbedPage renders as expected 1`] = `
<div
  className="tabbedContainer"
>
  <Tabs
    onClick={[Function]}
    onTab="info"
    tabs={Array []}
  />
  <TabsInfo
    info={Array []}
    onTab="info"
    tabs={Array []}
  />
</div>
`;
